packages/cli/lib/lib/webpack/prerender.js

```diff
@@ -100,39 +100,51 @@ async function handlePrerenderError(err, env, stack, entry) {
 
 	process.stderr.write('\n');
 	process.stderr.write(red(`${errorMessage}\n`));
-	process.stderr.write(`method: ${methodName}\n`);
-	if (sourceMapContent) {
+	// check if we have methodName (ie, the error originated in user code)
+	if (methodName) {
+		process.stderr.write(`method: ${methodName}\n`);
+		if (sourceMapContent & sourceCodeHighlight) {
+			process.stderr.write(
+				`at: ${sourcePath}:${position.line}:${position.column}\n`
+			);
+			process.stderr.write('\n');
+			process.stderr.write('Source code:\n\n');
+			process.stderr.write(sourceCodeHighlight);
+			process.stderr.write('\n');
+		} else {
+			process.stderr.write('\n');
+			process.stderr.write('Stack:\n\n');
+			process.stderr.write(JSON.stringify(stack, null, 4) + '\n');
+		}
+	} else {
 		process.stderr.write(
-			`at: ${sourcePath}:${position.line}:${position.column}\n`
+			yellow(
+				'Cannot determine error position. This most likely means it originated in node_modules.'
+			)
 		);
-		process.stderr.write('\n');
-		process.stderr.write('Source code:\n\n');
-		process.stderr.write(sourceCodeHighlight);
-		process.stderr.write('\n');
-	} else {
-		process.stderr.write(stack.toString() + '\n');
+		process.stderr.write('\n\n');
 	}
 	process.stderr.write(
 		`This ${
 			isReferenceError ? 'is most likely' : 'could be'
 		} caused by using DOM or Web APIs.\n`
 	);
 	process.stderr.write(
-		`Pre-render runs in node and has no access to globals available in browsers.\n\n`
+		'Pre-render runs in node and has no access to globals available in browsers.\n\n'
 	);
 	process.stderr.write(
-		`Consider wrapping code producing error in: 'if (typeof window !== "undefined") { ... }'\n`
+		'Consider wrapping code producing error in: "if (typeof window !== "undefined") { ... }"\n'
 	);
 
 	if (methodName === 'componentWillMount') {
-		process.stderr.write(`or place logic in 'componentDidMount' method.\n`);
+		process.stderr.write('or place logic in "componentDidMount" method.\n');
 	}
 	process.stderr.write('\n');
 	process.stderr.write(
-		`Alternatively use 'preact build --no-prerender' to disable prerendering.\n\n`
+		'Alternatively use "preact build --no-prerender" to disable prerendering.\n\n'
 	);
 	process.stderr.write(
-		'See https://github.com/developit/preact-cli#pre-rendering for further information.'
+		'See https://github.com/preactjs/preact-cli#pre-rendering for further information.\n\n'
```

Comment:
Interestingly enough, Yarn's (v1) own error message output will cover this last line without the added new line on the end.